cfgmgr/coppmgr.cpp

```diff
@@ -0,0 +1,745 @@
+                conf_present = true;
+            }
+
+            bool null_cfg = false;
+            for (auto i: kfvFieldsValues(t))
+            {
+                if (fvField(i) == COPP_TRAP_GROUP_FIELD)
+                {
+                    trap_group = fvValue(i);
+                }
+                else if (fvField(i) == COPP_TRAP_ID_LIST_FIELD)
+                {
+                    trap_ids = fvValue(i);
+                }
+                else if (fvField(i) == "NULL")
+                {
+                    null_cfg = true;
+                }
+            }
+            if (null_cfg)
+            {
+                if (!m_coppTrapConfMap[key].trap_group.empty())
+                {
+                    SWSS_LOG_DEBUG("Deleting trap key %s", key.c_str());
+                    removeTrapIdsFromTrapGroup(m_coppTrapConfMap[key].trap_group,
+                            m_coppTrapConfMap[key].trap_ids);
+                    trap_ids.clear();
+                    setCoppTrapStateOk(key);
+                    getTrapGroupTrapIds(m_coppTrapConfMap[key].trap_group, trap_ids);
+                    fvs.clear();
+                    FieldValueTuple fv(COPP_TRAP_ID_LIST_FIELD, trap_ids);
+                    fvs.push_back(fv);
+                    if (!checkTrapGroupPending(m_coppTrapConfMap[key].trap_group))
+                    {
+                        m_appCoppTable.set(m_coppTrapConfMap[key].trap_group, fvs);
+                    }
+                    setCoppGroupStateOk(m_coppTrapConfMap[key].trap_group);
+                    m_coppTrapConfMap[key].trap_group = "";
+                    m_coppTrapConfMap[key].trap_ids = "";
+                }
+                it = consumer.m_toSync.erase(it);
+                continue;
+            }
+            /*Duplicate check*/
+            if (conf_present &&
+                (trap_group == m_coppTrapConfMap[key].trap_group) &&
+                (trap_ids == m_coppTrapConfMap[key].trap_ids))
+            {
+                it = consumer.m_toSync.erase(it);
+                continue;
+            }
+            removeTrapIdsFromTrapGroup(m_coppTrapConfMap[key].trap_group,
+                                       m_coppTrapConfMap[key].trap_ids);
+            fvs.clear();
+            string trap_group_trap_ids;
+            addTrapIdsToTrapGroup(trap_group, trap_ids);
+            getTrapGroupTrapIds(trap_group, trap_group_trap_ids);
+            FieldValueTuple fv1(COPP_TRAP_ID_LIST_FIELD, trap_group_trap_ids);
+            fvs.push_back(fv1);
+            if (!checkTrapGroupPending(trap_group))
+            {
+                m_appCoppTable.set(trap_group, fvs);
+                setCoppGroupStateOk(trap_group);
+            }
+
+            /* When the trap table's trap group is changed, the old trap group
+             * should also be reprogrammed as some of its associated traps got
+             * removed
+             */
+            if ((!m_coppTrapConfMap[key].trap_group.empty()) &&
```

Comment:
Also check for `conf_present `before accessing the key.

Reply:
Done